tiledb/sm/consolidator/fragment_consolidator.cc

```diff
@@ -597,9 +597,10 @@ FragmentConsolidator::create_buffers(
       std::accumulate(buffer_weights.begin(), buffer_weights.end(), 0);
 
   // Allocate space for each buffer.
+  auto adjusted_budget = total_budget / total_weights * total_weights;
```

Comment:
Is this going to truncate, round down, do you maybe want it to round up?

Reply:
We want to round down to avoid issues below. The adjusted budget will always divide to a whole number when divided by the total weights. That way offsets buffer size will always be exactly 8 times larger than validity buffer size.

Reply:
I would suggest explicit type rather than `auto` for clarity when there are implications like this.